qiita_pet/handlers/analysis_handlers.py

```diff
@@ -321,3 +325,31 @@ def get(self):
                     user.shared_analyses + user.private_analyses]
 
         self.render("show_analyses.html", analyses=analyses)
+
+
+class ResultsHandler(StaticFileHandler, BaseHandler):
+    def validate_absolute_path(self, root, absolute_path):
+        """Overrides StaticFileHandler's method to include authentication
+        """
+        u = self.current_user
+
+        accessible_filepaths = get_accessible_filepath_ids(u)
+        accessible_filepaths = {join(get_db_files_base_dir(),
+                                     filepath_id_to_rel_path(fpid))
+                                for fpid in accessible_filepaths}
+
+        for accessible_filepath in accessible_filepaths:
+            access = False
+
+            if (accessible_filepath == absolute_path):
+                access = True
+
+            elif isdir(accessible_filepath) and \
+                    absolute_path.startswith(accessible_filepath + sep):
+                access = True
+
+            if access:
+                return super(ResultsHandler, self).validate_absolute_path(
+                    root, absolute_path)
+
+        raise QiitaPetAuthorizationError(u.id, absolute_path)
```

Comment:
A couple of issues with functionality:
- A regular user can not open its own studies. However, a user can not
open an admin study, which is fine.
- On a production site (with at least the EMP studies) these changes
make more evident this issue:
https://github.com/biocore/qiita/issues/684. Do you think is possible
to check for a specific file path id and user vs. getting all valid
file paths ids and then checking if the specific one is within that
group? This should make this faster. Note that this is specially bad
for admins as it returns basically all file path ids!